src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -74,18 +71,15 @@ public T set(T value, int index) {
     public T remove(int index) {
         Node<T> current = getNode(index);
         T oldValue = current.element;
-        if (size() == 1) {
-            head = null;
-            tail = null;
-        } else if (index == 0) {
-            current.next.prev = null;
-            head = current.next;
-        } else if (index == (size() - 1)) {
-            current.prev.next = null;
-            tail = current.prev;
-        } else {
+        if (current.prev != null && current.next != null) {
             current.prev.next = current.next;
             current.next.prev = current.prev;
+        } else if (current == tail && current.prev != null) {
+            current.prev.next = null;
+            tail = current.prev;
+        } else if (current == head && current.next != null) {
+            current.next.prev = null;
+            head = current.next;
```

Comment:
still too complicated. try to think this way: we found `current`, can we find `Node<T> prev` and `Node<T> next` and check their condition?